File: lista_de_exercicios/questao2/questao2.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self, incremento):
        self.velocidade += incremento

    def frear(self, decremento):
        self.velocidade = max(0, self.velocidade - decremento)

    def exibir_detalhes(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}"

    def exibir_velocidade(self):
        return f"Velocidade atual: {self.velocidade} km/h"

carro1 = Carro("Toyota", "Corolla", 2020)
carro1.acelerar(50)
print(carro1.exibir_velocidade())
carro1.frear(20)
print(carro1.exibir_velocidade())

File: lista_de_exercicios/questao2/questao2.go
package main
import "fmt"

type Carro struct {
    Marca     string
    Modelo    string
    Ano       int
    Velocidade int
}

func (c *Carro) Acelerar(incremento int) {
    c.Velocidade += incremento
}

func (c *Carro) Frear(decremento int) {
    if c.Velocidade >= decremento {
        c.Velocidade -= decremento
    } else {
        c.Velocidade = 0
    }
}

func (c Carro) ExibirDetalhes() string {
    return fmt.Sprintf("Marca: %s, Modelo: %s, Ano: %d", c.Marca, c.Modelo, c.Ano)
}

func (c Carro) ExibirVelocidade() string {
    return fmt.Sprintf("Velocidade atual: %d km/h", c.Velocidade)
}

func main() {
    carro1 := Carro{"Toyota", "Corolla", 2020, 0}
    carro1.Acelerar(50)
    fmt.Println(carro1.ExibirVelocidade())
    carro1.Frear(20)
    fmt.Println(carro1.ExibirVelocidade())
}
